### DISCUSSION TITLE
Scaling Milvus cluster

### DISCUSSION THREAD

**Discussion by nairan-deshaw** [milvus-io/milvus#34370] [Q&A and General discussion]
We currently have a cluster deployment of Milvus on K8s with close to 1TB in memory. There are about 16 query nodes each with 40G of memory. We're looking at data in the range of 250M vectors of 1536 dimension. Currently it uses IVF_FLAT indexing.
The main concerns are the slow partition load and release times before we run the queries. We are looking to through some more resources at the cluster and up the memory to about 3.5TB. What would be some recommendations for the number of query nodes and the memory assigned to each node. I've taken a look at the sizing tool but we also would like some recommendations here.

Some more related questions on this aspect:
- In addition, what would be a good index given we are looking for performance and a decent if not best accuracy. 
- We also came across  https://github.com/milvus-io/milvus/discussions/33659 and https://github.com/milvus-io/milvus/issues/25913 which briefly touch on these problems. Mainly we could see that increasing segment sizes and creating more in memory replicas of the partition should help. Can someone also help clarify these points?
- Is there some way to cache loaded partitions based on the access patterns which can help in reducing the overhead associated with loading and unloading the partitions every time.

**Comment by xiaofan-luan** [comment]
1. if you are looking for loading more data into main memory, diskANN or IVF_SQ8 might be a good option, IVF_SQ8 can reduce your memory to 1/4, and diskANN can reduce your memory usage to 1/10 with the help of local ssd.
2. For distributed deployment, we recommened each node is 16core64g, (too many cores may hit NUMA issue.) when you have 16 querynodes, for each collection create 4 shards is recommended.
3. increase segment size do help on the performance, i would recommend to use 2G or 4G segment size. Increase replica number also help on the performance.

**Comment by nairan-deshaw** [comment]
@xiaofan-luan question around the querynodes:

Is it recommended to have multiple querynodes with smaller memory or less querynodes with larger memory? 
Say with a 100G of memory does 4 replicas with 25G each perform better than 10 replicas with 10G each? Also, what be the reasons why either is better?

**Reply by xiaofan-luan** [reply]
the recommended memory size for each querynode is 8core 32G/64G ,16 core 64G/128G.

CPU core number should not exceed numa boundary, but node should not be too small either
